### PR TITLE
Add union function to HashMap

### PR DESCRIPTION
**Pull request by 7sharp9**
Pretty much the same as the Map.union function

### CODE REVIEW

File: ExtCore/Collections.HashMap.fs [hunk 1]
```diff
@@ -2042,3 +2042,30 @@ module HashMap =
...
+    /// </summary>
+    /// <param name="map1"></param>
+    /// <param name="map2"></param>
+    /// <returns></returns>
+    [<CompiledName("Union")>]
+    let union (map1 : HashMap<'Key, 'T>) (map2 : HashMap<'Key, 'T>) :  HashMap<'Key, 'T> =
+        checkNonNull "map1" map1
+        checkNonNull "map2" map2
+        
+        match   count map1, count map2 with
```
Comment (jack-pappas) [inline ExtCore/Collections.HashMap.fs]: ``count`` needs to traverse the map in order to determine the number of key-value pairs in the map. Given you're only concerned with whether the map is empty (or not), please use ``isEmpty`` for that check instead (it's an _O(1)_ operation).

**Comment by jack-pappas** [comment]
Thanks Dave! Would you mind adding a basic unit test or two for this function as well?

**Comment by 7sharp9** [comment]
@jack-pappas Yep, Ill add a few tests.  Incidentally I was using this to try and improve on the map performance of some HM type inference with row polymorphism.  Except it seems to be slower part from high iteration counts vs map.  Did you ever look into PersistantHashMap?

**Comment by jack-pappas** [comment]
PersistentHashMap as in the Scala or Clojure 'vector' type? I did look at it at one point but never got around to actually implementing it for ExtCore.

If you have an example / benchmark of some code where ``HashMap`` is slower than F# ``Map`` and you can post it somewhere, I'm happy to take a look. ``HashMap`` isn't *guaranteed* to be faster, but empirically I have found it to be faster in most cases I've used it.

**Comment by 7sharp9** [comment]
@jack-pappas Im implementing various HM type inference algorithms with extension like row and variant polymorphism, possibly rank n types as well.  I found the 100,000 iterations was the time when map was outperformed by hashmap.  To be honest I was using a string as a type var identifier where an int and rank would be more efficient.  Even not got to the point where Im overly worried about performance yet, Im just very mindful of the generic comparison overhead on map, especially with a string as I was using, I was surprised hashmap was slower for lower iterations.

**Pull request merged**